### PR TITLE
Signaling support

### PR DESCRIPTION
**Pull request by JayTokBox**
All connections and specific connections with rspec tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -178,6 +178,28 @@ as the `:archive_mode` property of the `options` parameter passed into the
 For more information on archiving, see the
 [OpenTok archiving](https://tokbox.com/opentok/tutorials/archiving/) programming guide.
 
+## Signaling
+
+You can send a signal using the `opentok.signal.send(session_id, connection_id, opts)` method.  
+If `connection_id` is nil or an empty string, then the signal is send to all valid connections in the 
+session. 
+
+The hash `opts` field is optional. An example of it is as follows:
```
Comment (normanargueta) [inline README.md]: The opts field is optional?

File: lib/opentok/opentok.rb [hunk 2]
```diff
@@ -177,8 +177,11 @@ def sip
       @sip ||= Sip.new client
     end
 
-    protected
+    def signal
+      @signal ||= Signals.new client
```
Comment (normanargueta) [inline lib/opentok/opentok.rb]: Are you missing the require of the "Signals" class (require "opentok/signals") in this file?

File: lib/opentok/signals.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    # session.
+    #
+    # For more information on signaling, see the
+    # {https://tokbox.com/developer/rest/#send_signal} programming guide.
+    #
+    # @param [String] session_id The session ID of the OpenTok session.
+    # @param [Hash] options  A hash with the key 'type', 'data', and 'connId'.
+    # @option options [String] :type This is the type of the signal. You can use this
+    # field to group and filter signals. It is a property of the Signal object.
+    # @option options [String] :data This is the data within the the signal or the payload.
```
Comment (normanargueta) [inline lib/opentok/signals.rb]: Typo error "the the signal"

File: lib/opentok/signals.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    # Contains main information to be sent in the signal. It is a property of the Signal object.
+    # @option options [String] :connId When a connId is specified only that connection recieves a signal.
```
Comment (normanargueta) [inline lib/opentok/signals.rb]: According to the documentation the only options in the hash should be "type" and "data"

File: README.md (hunk 1 above)
Comment (JayTokBox) [reply README.md]: Theoretically yes , but it is a rare case.  Hence I am rewording it as "An example of `opts` field can be as follows:"

File: lib/opentok/signals.rb (hunk 4 above)
Comment (JayTokBox) [reply lib/opentok/signals.rb]: good catch !! thx

**Pull request merged**